stdlib/2and3/ctypes/__init__.pyi

```diff
@@ -0,0 +1,257 @@
+    def WINFUNCTYPE(restype: Type[_CData],
+                    *argtypes: Type[_CData],
+                    use_errno: bool = ...,
+                    use_last_error: bool = ...) -> Type[_FuncProto]: ...
+def PYFUNCTYPE(restype: Type[_CData],
+               *argtypes: Type[_CData]) -> Type[_FuncProto]: ...
+
+_PF = UnionT[
+    Tuple[int],
+    Tuple[int, str],
+    Tuple[int, str, Any]
+]
+
+class _FuncProto(_FuncPtr):
+    @overload
+    def __init__(self, address: int) -> None: ...
+    @overload
+    def __init__(self, callable: Callable[..., Any]) -> None: ...
+    @overload
+    def __init__(self, func_spec: Tuple[UnionT[str, int], _DLL],
+                 paramflags: Tuple[_PF, ...] = ...) -> None: ...
+    @overload
+    def __init__(self, vtlb_index: int, name: str,
+                 paramflags: Tuple[_PF, ...] = ...,
+                 iid: _Pointer[c_int] = ...) -> None: ...
+
+class _cparam: ...
+
+def addressof(obj: _CData) -> int: ...
+def alignment(obj_or_type: UnionT[_CData, Type[_CData]]) -> int: ...
+def byref(obj: _CData, offset: int = ...) -> _cparam: ...
+_PT = TypeVar('_PT', bound=_PointerLike)
+def cast(obj: _CData, type: Type[_PT]) -> _PT: ...
+def create_string_buffer(init_or_size: UnionT[int, bytes],
+                         size: Optional[int] = ...) -> Array[c_char]: ...
+c_buffer = create_string_buffer
+def create_unicode_buffer(init_or_size: UnionT[int, str],
+                          size: Optional[int] = ...) -> Array[c_wchar]: ...
+if sys.platform == 'win32':
+    def DllCanUnloadNow() -> int: ...
+    def DllGetClassObject(rclsid: Any, riid: Any, ppv: Any) -> int: ...  # TODO not documented
+    def FormatError(code: int) -> str: ...
+    def GetLastError() -> int: ...
+def get_errno() -> int: ...
+if sys.platform == 'win32':
+    def get_last_error() -> int: ...
+def memmove(dst: UnionT[int, _CData],
+            src: UnionT[int, _CData],
+            count: int) -> None: ...
+def memset(dst: UnionT[int, _CData],
+           c: int, count: int) -> None: ...
+def POINTER(type: Type[_CT]) -> Type[_Pointer[_CT]]: ...
+def pointer(obj: _CT) -> _Pointer[_CT]: ...
+def resize(obj: _CData, size: int) -> None: ...
+if sys.version_info < (3,):
+    def set_conversion_mode(encoding: str, errors: str) -> Tuple[str, str]: ...
+def set_errno(value: int) -> int: ...
+if sys.platform == 'win32':
+    def set_last_error(value: int) -> int: ...
+def sizeof(obj_or_type: UnionT[_CData, Type[_CData]]) -> int: ...
+def string_at(address: int, size: int = ...) -> bytes: ...
+if sys.platform == 'win32':
+    def WinError(code: Optional[int] = ...,
+                 desc: Optional[str] = ...) -> OSError: ...
+def wstring_at(address: int, size: int = ...) -> str: ...
+
+class _SimpleCData(Generic[_T], _CData):
+    value: _T = ...
+    def __init__(self, value: _T = ...) -> None: ...
+
+class c_byte(_SimpleCData[int]): ...
+
+class c_char(_SimpleCData[bytes]): ...
+class c_char_p(_PointerLike, _SimpleCData[Optional[bytes]]):
```

Comment:
Why is it Optional? If the value is inaccessible, it just segfaults (as I just found out experimentally).

Reply:
Not here - the `value` of a null pointer is `None`.

```python
Python 3.6.4 (v3.6.4:d48ecebad5, Dec 18 2017, 21:07:28) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
>>> import ctypes
>>> ccp = ctypes.c_char_p(None)
>>> ccp.value
>>> cwp = ctypes.c_wchar_p(None)
>>> cwp.value
```